### PR TITLE
Add support for Mica effect on Windows 11

### LINKED ISSUE #6465: Acrylic Blur does not work on Windows 11
**Describe the bug**
Win2d acrylic blur does not work on Windows 11 build 22000.160. This also breaks `TransparencyLevel.Blur` if WinUI compositor is enabled.

DWM transparency and blur works fine.

**To Reproduce**

1. Create new project from template
2. Set `TransparencyLevelHint.AcrylicBlur` on the main window.
3. Set background to `null`.
4. Run

This can also be observed in the SampleApp.

**Expected behavior**
The acrylic layer should not be opaque.

**Screenshots**
image

**Desktop (please complete the following information):**
- OS: Windows 11 22000.160
- Version 0.10.3; 0.10.7 (master)

**Additional context**
Side note: `ICompositor3.CreateHostBackdropBrush()` is _not_ broken in this build of windows. `ICompositor2.CreateBackdropBrush()` might be though.

### PR DESCRIPTION
**Pull request by adirh3**
## What does the pull request do?
Add support for Windows 11 new backdrop brush.
Add support for creating backdrop brushes with rounded corners on Windows 10/11.
Popups with blur effect on them setting (now that rounded corner blur is working) - 
```
 <Style Selector="PopupRoot">
            <Setter Property="TransparencyLevelHint" Value="AcrylicBlur" />
  </Style>          
```

## What is the current behavior?
Mica effect is not available on Windows 11 (even if user wants to implement it outside Avalonia)
Can't create true rounded corner window on Windows 10 with any blur effect.

## What is the updated/expected behavior with this PR?
Users will be able to create rounded corner applications on Windows 10 and use the new Mica effect on Windows 11.
image
This is already available in my app, if you'd like to test here is a portable version - https://install.appcenter.ms/users/adirh3-gmail.com/apps/fluent-search-alpha/distribution_groups/portable

## How was the solution implemented (if it's not obvious)?
Added support explicitly only on Windows 11.
I understood that in the future Avalonia will have built-in renderer with layering support, which will make this possible to implement in a cross-platform way and not relay on Windows composition APIs.

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Documentation with user documentation

## Fixed issues
Fixes #6465

### CODE REVIEW

File: src/Avalonia.Controls/WindowTransparencyLevel.cs [hunk 1]
```diff
@@ -20,6 +20,11 @@ public enum WindowTransparencyLevel
         /// <summary>
         /// The window background is a blur-behind with a high blur radius. This level may fallback to Blur.
         /// </summary>
-        AcrylicBlur
+        AcrylicBlur,
+        
+        /// <summary>
+        /// The window background is based on desktop wallpaper tint with a blur. This will only work on Windows 11 
+        /// </summary>
+        Mica
```
Comment (maxkatz6) [inline src/Avalonia.Controls/WindowTransparencyLevel.cs:28]: I am not sure about having very platform specific value in here.
Even acrylic exists at least in macOS.

Comment (maxkatz6) [reply src/Avalonia.Controls/WindowTransparencyLevel.cs:28]: AFAIK AcrylicBlur doesn't work rn in Win11. Was it completely replaced by Mica?

Comment (adirh3) [reply src/Avalonia.Controls/WindowTransparencyLevel.cs:28]: @maxkatz6 I know it's a little weird, but right now `WindowTransparencyLevel` is not really cross-platform, the actual value used by each `WindowImpl` is different depending on the situation.
Also, I think that this effect can be implemented in the future as a cross platform effect once new renderer will be available.
As for AcrylicBlur on Win11, it works if you don't use `BorderOnly` as WindowDecoration (which call some DWM API that was changed to creates a static accent color on window background).

Comment (kekekeks) [reply src/Avalonia.Controls/WindowTransparencyLevel.cs:28]: TBH should have separated transparency from backdrop effects.
1) gracefull fallback to regular transparency if blur effects aren't available isn't possible with the current API
2) if we keep adding more stuff to that enum it becomes more and more weird

I think I'm ok with adding `Mica` as enum member for 0.10.x timeframe, but for the next version we need separate  `PreferredBackdropDecorationEffects` (`None`, `AcryllicBlur`, `Mica`, `Blur`) and `WantsTransparency` (`true`, `false`, takes effect when effective backdrop decoration is `None`)

File: src/Windows/Avalonia.Win32/Win32Platform.cs [hunk 2]
```diff
@@ -89,6 +89,12 @@ public class Win32PlatformOptions
         /// This is recommended if you need to use AcrylicBlur or acrylic in your applications.
         /// </remarks>
         public bool UseWindowsUIComposition { get; set; } = true;
+
+        /// <summary>
+        /// When <see cref="UseWindowsUIComposition"/> enabled, create rounded corner blur brushes
+        /// If set to zero the brushes will be created using default settings (sharp corners)
+        /// </summary>
+        public float CompositionBackdropCornerRadius { get; set; }
```
Comment (maxkatz6) [inline src/Windows/Avalonia.Win32/Win32Platform.cs:97]: Does it mean that by default avalonia apps won't have rounded corners?

Comment (adirh3) [reply src/Windows/Avalonia.Win32/Win32Platform.cs:97]: Yes, sharp corners just as it's now. 
I am still thinking whether to add this as a window property but then that might be only possible using something similar to what @kekekeks suggested above, do you think it's worth implementing as part of this PR?

Comment (maxkatz6) [reply src/Windows/Avalonia.Win32/Win32Platform.cs:97]: I am sorry for missing this question. 
I don't like that this property is missleading. It says "If set to zero the brushes will be created using default settings (sharp corners)", but in reality default setting now depends on windows OS - it's sharp on 10 and round on 11. Is there any way to get its default value from the API?
We can keep this property nullable and apply value only when it's not null.

**Review by maxkatz6 — APPROVED** [review]

**Pull request merged**